src/validators/literal.rs

```diff
@@ -31,12 +38,23 @@ pub struct LiteralLookup<T: Clone + Debug> {
 impl<T: Clone + Debug> LiteralLookup<T> {
     pub fn new<'py>(py: Python<'py>, expected: impl Iterator<Item = (&'py PyAny, T)>) -> PyResult<Self> {
         let mut expected_int = AHashMap::new();
-        let mut expected_str = AHashMap::new();
+        let mut expected_str: AHashMap<String, usize> = AHashMap::new();
+        let mut expected_bool = BoolLiteral {
+            true_id: None,
+            false_id: None,
+        };
         let expected_py = PyDict::new(py);
         let mut values = Vec::new();
         for (k, v) in expected {
             let id = values.len();
             values.push(v);
+            if let Ok(bool) = k.strict_bool() {
+                if bool {
+                    expected_bool.true_id = Some(id);
+                } else {
+                    expected_bool.false_id = Some(id);
+                }
+            }
```

Comment:
The chain of `if let Ok(...) = input.x()` here just seems to scream at me for the input-as-enum we spoke about the other day...

Reply:
Agreed!